src/libraries/Common/src/Interop/Interop.TimeZoneInfo.cs

```diff
@@ -16,9 +16,9 @@ internal static partial class Globalization
             int resultLength);
 
         [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_WindowsIdToIanaId")]
-        internal static extern unsafe int WindowsIdToIanaId(string windowsId, char* ianaId, int ianaIdLength);
+        internal static extern unsafe int WindowsIdToIanaId(string windowsId, string? region, ref char ianaId, int ianaIdLength);
 
         [DllImport(Libraries.GlobalizationNative, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_IanaIdToWindowsId")]
-        internal static extern unsafe int IanaIdToWindowsId(string ianaId, char* windowsId, int windowsIdLength);
+        internal static extern unsafe int IanaIdToWindowsId(string ianaId, ref char windowsId, int windowsIdLength);
```

Comment:
I didn't think we were supposed to use `ref char` in P/Invoke signatures because it isn't just a single char being referenced, but a whole sequence of chars.

Reply:
The address of the first character is the same as the address of the whole string. I can change this to char* but that will make the calling site unsafe which I am not seeing a reason we need to do so. 

@jkotas may advise about that.

Reply:
Using `ref char windowsId` to represent address of a string is even more unsafe pattern than `char*`. I have a strong preference for using `char*` in cases like this one.

> make the calling site unsafe 

You are using `MemoryMarshal.GetReference` at the callsite that is pretty unsafe even though the C# does not require unsafe C# for it.

Reply:
Yes, I meant not intentionally marking the code `unsafe`. The question now, do you have any concern using `ref char` for a stack buffer?

Reply:
I'll change back to `char*` anyway. Thanks all.